src/com/ms/silverking/cloud/dht/daemon/storage/convergence/management/NamespaceRequest.java

```diff
@@ -17,81 +17,81 @@
 import com.ms.silverking.time.Timer;
 
 /**
- * Tracks outstanding requests for namespaces from peers. 
+ * Tracks outstanding requests for namespaces from peers.
  */
 class NamespaceRequest {
-    private final MessageGroupBase  mgBase;
-    private final Set<IPAndPort>    incompletePeers;
-    private final Map<UUIDBase,NamespaceRequest>  nsRequests;
-    private final Set<Long>            namespaces;
-    
-    private static final boolean    debug = false;
-    
-    NamespaceRequest(MessageGroupBase mgBase, Set<IPAndPort> peers, Map<UUIDBase,NamespaceRequest> nsRequests) {
-        this.mgBase = mgBase;
-        this.incompletePeers = new HashSet<>(peers);
-        this.nsRequests = nsRequests;
-        this.namespaces = new ConcurrentSkipListSet<>();
-    }
-    
-    public boolean waitForCompletion(int waitLimitMillis) {
-        synchronized (incompletePeers) {
-            Timer    timer;
-            
-            timer = new SimpleTimer(TimeUnit.MILLISECONDS, waitLimitMillis);
-            while (incompletePeers.size() > 0 && !timer.hasExpired()) {
-                try {
-                    incompletePeers.wait(timer.getRemainingMillis());
-                } catch (InterruptedException ie) {
-                }
-            }
-            if (incompletePeers.size() > 0) {
-                Log.warning("Unable to receive namespaces from: "+ CollectionUtil.toString(incompletePeers));
-            }
-            return incompletePeers.size() == 0;
+  private final MessageGroupBase mgBase;
+  private final Set<IPAndPort> incompletePeers;
+  private final Map<UUIDBase, NamespaceRequest> nsRequests;
+  private final Set<Long> namespaces;
+
+  private static final boolean debug = false;
+
+  NamespaceRequest(MessageGroupBase mgBase, Set<IPAndPort> peers, Map<UUIDBase, NamespaceRequest> nsRequests) {
+    this.mgBase = mgBase;
+    this.incompletePeers = new HashSet<>(peers);
+    this.nsRequests = nsRequests;
+    this.namespaces = new ConcurrentSkipListSet<>();
+  }
+
+  public boolean waitForCompletion(int waitLimitMillis) {
+    synchronized (incompletePeers) {
+      Timer timer;
+
+      timer = new SimpleTimer(TimeUnit.MILLISECONDS, waitLimitMillis);
+      while (incompletePeers.size() > 0 && !timer.hasExpired()) {
+        try {
+          incompletePeers.wait(timer.getRemainingMillis());
+        } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)